ipykernel/kernelbase.py

```diff
@@ -1007,13 +1021,35 @@ def do_is_complete(self, code):
         return {"status": "unknown"}
 
     async def debug_request(self, socket, ident, parent):
-        """Handle a debug request."""
+        """Handle a debug request.
+
+        Seem Deprecated
+        """
+        # If this is incorrect, remove `debug_request` from the deprecated message types
+        # at the beginning of this class
+        warnings.warn(
+            "`debug_request` is deprecated in `kernelbase.py` since ipykernel 6.10"
+            " (2022). It is only part of IPython parallel. Did you mean do_debug_request ?",
+            DeprecationWarning,
+            stacklevel=2,
+        )
         if not self.session:
             return
         content = parent["content"]
-        reply_content = self.do_debug_request(content)
-        if inspect.isawaitable(reply_content):
-            reply_content = await reply_content
+        if not inspect.iscoroutinefunction(self.do_debug_request):
+            # repeat the warning at run
+            reply_content = self.do_debug_request(content)
+            warnings.warn(
```

Comment:
Not deprecated. This is part of the debugger protocol, not part of IPython Parallel